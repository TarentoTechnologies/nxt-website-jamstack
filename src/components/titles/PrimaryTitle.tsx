import { white } from "../../styles/style-guide/ColorGuide.module.css";
import {
  acceleratorTitle,
  cardHeader,
  h1One,
  h1700,
  h1700V2,
  h1700V3,
  h3600,
  heroTitle,
  infoSectionH2,
  infoSectionH3,
} from "../../styles/style-guide/Typography.module.css";

interface PrimaryTitleProps {
  title: string;
  variant?: string;
  color?: string;
  customClasses?: any;
}

export const PrimaryTitle = ({
  title,
  variant,
  color,
  customClasses,
}: PrimaryTitleProps) => {
  return (
    <div
      className={`
      ${customClasses}
        ${variant === "banner" ? h1One : ""}
        ${variant === "heroTitle" ? heroTitle : ""}
        ${variant === "infoSectionH2" ? infoSectionH2 : ""}
        ${variant === "infoSectionH3" ? infoSectionH3 : ""}
        ${variant === "card" ? cardHeader : ""}
        ${variant === "accelerator" ? acceleratorTitle : ""}
        ${variant === "h3600" ? h3600 : ""}
        ${variant === "h1700" ? h1700 : ""}
        ${variant === "h1700V2" ? h1700V2 : ""}
        ${variant === "h1700V3" ? h1700V3 : ""}
        ${color === "white" ? white : ""}`}
    >
      {title}
    </div>
  );
};
