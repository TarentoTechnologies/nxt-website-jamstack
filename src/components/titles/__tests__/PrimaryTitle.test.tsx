import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { PrimaryTitle } from "../PrimaryTitle";

describe("Test if the PrimaryTitle component is working as expected", () => {
  it("<PrimaryTitle /> matches snapshot", () => {
    const component = render(
      <PrimaryTitle data-testid="primary-title" title={"Case Studies"} />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<PrimaryTitle /> renders without exploding", () => {
    const component = render(
      <PrimaryTitle data-testid="primary-title" title={"Case Studies"} />
    );
    expect(component.getByTestId("primary-title")).toBeInTheDocument();
  });
});
