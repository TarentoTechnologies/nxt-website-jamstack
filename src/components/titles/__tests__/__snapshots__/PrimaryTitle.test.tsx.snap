// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the PrimaryTitle component is working as expected <PrimaryTitle /> matches snapshot 1`] = `
<div>
  <div
    class="
      undefined
        
        
        
        
        
        
        
        
        
        
        "
    data-aos="fade-up"
    data-testid="primary-title"
  >
    Case Studies
  </div>
</div>
`;
